client/src/js/services/FilterService.js

```diff
@@ -4,50 +4,183 @@ angular.module('bhima.services')
+    this._filterIndex[key].assign(null, null);
+  }
+
+  FilterList.prototype._resetCustomFilters = function resetCustomFilters() {
+    this._filterActiveFilters().forEach(function (filter) {
+      // only by default remove custom values
+      if (!filter._isDefault) {
+        this.resetFilterState(filter._key);
+      }
+    }.bind(this));
+  }
+
+  // @TODO registerDefaultFilter and registerCustomFilter could use the same underlying function
+  //       with a toggle between the array to populate and the default value
+  FilterList.prototype.registerDefaultFilters = function registerDefaultFilters(filterDefinitions) {
+    var formattedFilters = filterDefinitions.map(function (filterDefinition) {
+      var filter = new Filter(filterDefinition.key, filterDefinition.label, filterDefinition.valueFilter);
+      filter.setDefault(true);
+
+      if (filterDefinition.defaultValue) {
+        filter.assign(filterDefinition.defaultValue);
+      }
+      return filter;
+    });
+
+    // udpate index
+    this._indexList(this._filterIndex, formattedFilters);
+    this._defaultFilters = this._defaultFilters.concat(formattedFilters);
+  };
+
+  FilterList.prototype.registerCustomFilters = function registerCustomFilters(filterDefinitions) {
+    var formattedFilters = filterDefinitions.map(function (filterDefinition) {
+      var filter = new Filter(filterDefinition.key, filterDefinition.label, filterDefinition.valueFilter);
+      filter.setDefault(false);
+      return filter;
+    });
+
+    // udpate index
+    this._indexList(this._filterIndex, formattedFilters);
+    this._customFilters = this._customFilters.concat(formattedFilters);
+  };
+
+
+  // assigns the value of a filter, a filter with a value will be actively used
+  // during the HTTP/ UI export process
+  FilterList.prototype.assignFilter = function assignFilter(key, value, displayValue) {
+    this._filterIndex[key].assign(value, displayValue);
+  };
+
+  // accepts an array of key : filterValue objects that are assigned
+  // [
+  // { key : value },
+  // { key : value }
+  // ]
+  FilterList.prototype.assignFilters = function assignFilters(valueList) {
+    valueList.forEach(function (valueMap) {
+      this.assignFilter(valueMap.key, valueMap.value, valueMap.displayValue);
+    }.bind(this));
+  };
+
+  // alias for `assignFilters`, clears the currently active filters before
+  // calling the erferenced method
+  FilterList.prototype.replaceFilters = function replaceFilters(valueList) {
+    this._resetCustomFilters();
+    this.assignFilters(valueList);
+  };
+
+  // return filters for the view - this method will always be compatible with the bhFilter component
+  FilterList.prototype.formatView = function formatView() {
+    var activeFilters = this._filterActiveFilters();
+    var activeKeys = activeFilters.map(function (filter) {  return filter._key });
+
+    function keysInActive(filter) { return activeKeys.includes(filter._key); }
```

Comment:
[Array.prototype.includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) is ECMAScript version 6.  I think it's probably best to avoid it until we can compile ES6 on the client (even if it is supported by Chrome).  What do you think?